lib/auth/auth_with_roles.go

```diff
@@ -4542,7 +4542,7 @@ func (a *ServerWithRoles) GetSAMLIdPSession(ctx context.Context, req types.GetSA
 	}
 	// Users can only fetch their own SAML IdP sessions.
 	if err := a.currentUserAction(session.GetUser()); err != nil {
```

Comment:
Why are we doing the RBAC check _after_ we've fetched the session?

Reply:
Wait, does this mean that in this and `GetAppSession`/`GetSnowflakeSession` we're actually revealing if a session exists even if the requester doesn't have permissions to read it?

Reply:
Yes it does. IIRC I tried to fix this before, but I can't remember what the conversation was at the time. Want me to make this error opaque while I'm here?

Reply:
Ahhh, I think I know what it was. I think there are things that depend on the `NotFound` return from sessions that makes fixing this non-trivial.

Reply:
It's this: https://github.com/gravitational/teleport/pull/22586 that I was thinking of. :-) Not the same thing, but I think I maintained the semantics for a similar reason.